File: frontend/src/utils/unusedReason.ts
export const UNUSED_REASON_OPTIONS = [
  { value: "purchase_cancelled", label: "Alımdan Vazgeçildi." },
  { value: "no_longer_needed", label: "İhtiyaç Kalmadı." },
  { value: "other_budget", label: "Başka Bütçeden Karşılandı." },
  { value: "unused", label: "Kullanılmayacak." }
] as const;

const legacyLabels: Record<string, string> = {
  "Alımdan vazgeçildi": "purchase_cancelled",
  "İhtiyaç kalmadı": "no_longer_needed",
  "Başka Bütçe": "other_budget",
  Kullanılmayacak: "unused"
};

export function normalizeUnusedReason(value?: string | null): string {
  if (!value) return "";
  return UNUSED_REASON_OPTIONS.find((option) => option.value === value || option.label === value)?.value
    ?? legacyLabels[value]
    ?? "";
}

export function formatUnusedReason(value?: string | null, fallback = "-"): string {
  const normalized = normalizeUnusedReason(value);
  return UNUSED_REASON_OPTIONS.find((option) => option.value === normalized)?.label ?? fallback;
}

File: app/services/unused_reason.py
UNUSED_REASON_LABELS = {
    "purchase_cancelled": "Alımdan Vazgeçildi.",
    "no_longer_needed": "İhtiyaç Kalmadı.",
    "other_budget": "Başka Bütçeden Karşılandı.",
    "unused": "Kullanılmayacak.",
}

UNUSED_REASON_ALIASES = {
    **{code: code for code in UNUSED_REASON_LABELS},
    **{label: code for code, label in UNUSED_REASON_LABELS.items()},
    "Alımdan vazgeçildi": "purchase_cancelled",
    "İhtiyaç kalmadı": "no_longer_needed",
    "Başka Bütçe": "other_budget",
    "Kullanılmayacak": "unused",
}


def unused_reason_label(value: str | None) -> str:
    if not value:
        return ""
    code = UNUSED_REASON_ALIASES.get(value, value)
    return UNUSED_REASON_LABELS.get(code, "")
